repos/exporter/lib/exporter-stack.ts

```diff
@@ -1,59 +1,70 @@
-import { Bucket, BucketEncryption } from '@aws-cdk/aws-s3';
+import { Bucket, BucketEncryption, IBucket } from '@aws-cdk/aws-s3';
 import { NodejsFunction, SourceMapMode } from '@aws-cdk/aws-lambda-nodejs';
 import { PolicyStatement } from '@aws-cdk/aws-iam';
-import { SecurityGroup, Subnet, Vpc } from '@aws-cdk/aws-ec2';
+import { SecurityGroup, Subnet, Vpc, ISecurityGroup, IVpc, ISubnet } from '@aws-cdk/aws-ec2';
 import { Topic } from '@aws-cdk/aws-sns';
 import { SnsAction } from '@aws-cdk/aws-cloudwatch-actions';
 import { TreatMissingData } from '@aws-cdk/aws-cloudwatch';
 import { Rule, Schedule } from '@aws-cdk/aws-events';
 import { LambdaFunction } from '@aws-cdk/aws-events-targets';
 
 export class ExporterStack extends cdk.Stack {
+    private readonly stage: string;
+    private readonly matchingDataBucket: IBucket;
+    private readonly productsBucket: IBucket;
+
     constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
         super(scope, id, props);
 
         const stage = process.env.STAGE;
         if (!stage) {
             throw new Error('Need to set STAGE');
         }
+        this.stage = stage;
+
+        const vpc = Vpc.fromLookup(this, 'vpc', { vpcName: `fdbt-vpc-${this.stage}` });
+        const vpcSubnets = {
+            subnets: [
+                Subnet.fromSubnetId(this, 'vpc-subnet-a', Fn.importValue(`${this.stage}:PrivateSubnetA`)),
+                Subnet.fromSubnetId(this, 'vpc-subnet-b', Fn.importValue(`${this.stage}:PrivateSubnetB`)),
+            ],
+        };
+        const securityGroup = SecurityGroup.fromSecurityGroupId(
+            this,
+            'security-group',
+            Fn.importValue(`${this.stage}:ReferenceDataUploaderLambdaSG`),
+        );
 
-        const productsBucket = new Bucket(this, `products-data-bucket-${stage}`, {
-            bucketName: `fdbt-products-data-${stage}`,
+        this.productsBucket = new Bucket(this, `products-data-bucket-${this.stage}`, {
+            bucketName: `fdbt-products-data-${this.stage}`,
             encryption: BucketEncryption.KMS_MANAGED,
         });
 
-        const matchingDataBucket = Bucket.fromBucketName(
+        this.matchingDataBucket = Bucket.fromBucketName(
             this,
-            `fdbt-matching-data-${stage}`,
-            `fdbt-matching-data-${stage}`,
+            `fdbt-matching-data-${this.stage}`,
+            `fdbt-matching-data-${this.stage}`,
         );
 
-        const vpc = Vpc.fromLookup(this, 'vpc', { vpcName: `fdbt-vpc-${stage}` });
-        const vpcSubnets = {
-            subnets: [
-                Subnet.fromSubnetId(this, 'vpc-subnet-a', Fn.importValue(`${stage}:PrivateSubnetA`)),
-                Subnet.fromSubnetId(this, 'vpc-subnet-b', Fn.importValue(`${stage}:PrivateSubnetB`)),
-            ],
-        };
+        this.addNeedsAttentionLambda(securityGroup, vpc, vpcSubnets);
+        this.addReferenceDataLambdas(securityGroup, vpc, vpcSubnets);
```

Comment:
Does it make sense for these to all be in the same CDK Stack?

They seem to be distinct enough parts of the application, otherwise we're heading in the direction of the `ExporterStack` really being the everything stack, we should look to set a good pattern on this from the beginning

Reply:
Yeah I've been meaning to rename the stack before I leave now it does more stuff. I'll try and get this done separately